query-node/mappings/src/content/commentAndReaction.ts

```diff
@@ -0,0 +1,708 @@
+  const { memberId, commentId, reactionId } = idSegments
+  return `${memberId}-${commentId}-${reactionId}`
+}
+
+async function getOrCreateVideoReactionsCountByReactionId(
+  store: DatabaseManager,
+  video: Video,
+  reaction: VideoReactionOptions
+): Promise<VideoReactionsCountByReactionType> {
+  const reactionsCountByReactionId = await store.get(VideoReactionsCountByReactionType, {
+    where: { id: `${video.id}-${reaction}` },
+  })
+
+  if (!reactionsCountByReactionId) {
+    const newReactionsCountByReactionId = new VideoReactionsCountByReactionType({
+      id: `${video.id}-${reaction}`,
+      reaction,
+      video,
+      count: 0,
+    })
+
+    await store.save<VideoReactionsCountByReactionType>(newReactionsCountByReactionId)
+    return newReactionsCountByReactionId
+  }
+
+  return reactionsCountByReactionId
+}
+
+async function getOrCreateCommentReactionsCountByReactionId(
+  store: DatabaseManager,
+  comment: Comment,
+  reactionId: number
+): Promise<CommentReactionsCountByReactionId> {
+  const reactionsCountByReactionId = await store.get(CommentReactionsCountByReactionId, {
+    where: { id: `${comment.id}-${reactionId}` },
+  })
+
+  if (!reactionsCountByReactionId) {
+    const newReactionsCountByReactionId = new CommentReactionsCountByReactionId({
+      id: `${comment.id}-${reactionId}`,
+      reactionId,
+      comment,
+      count: 0,
+    })
+
+    await store.save<CommentReactionsCountByReactionId>(newReactionsCountByReactionId)
+    return newReactionsCountByReactionId
+  }
+
+  return reactionsCountByReactionId
+}
+
+function parseVideoReaction(reaction: ReactVideo.Reaction): VideoReactionOptions {
+  switch (reaction) {
+    case ReactVideo.Reaction.LIKE: {
+      return VideoReactionOptions.LIKE
+    }
+    case ReactVideo.Reaction.UNLIKE: {
+      return VideoReactionOptions.UNLIKE
+    }
+  }
+}
+
+export async function processReactVideoMessage(
+  { store, event }: EventContext & StoreContext,
+  memberId: MemberId,
+  message: IReactVideo
+): Promise<void> {
+  const { videoId, reaction } = message
+  const eventTime = new Date(event.blockTimestamp)
+  const reactionResult = parseVideoReaction(reaction)
+
+  // load video
+  const video = await getVideo(store, videoId.toString(), ['channel', 'channel.bannedMembers'])
+
+  // ensure member is not banned from channel
+  ensureMemberIsNotBannedFromChannel(video.channel, memberId.toString(), 'Cannot add reaction')
+
+  // ensure reaction feature is enabled on the video
+  ensureReactionFeatureIsEnabled(video)
+
+  // load previous reaction by member to the video (if any)
+  const previousReactionByMember = await store.get(VideoReaction, {
+    where: { id: videoReactionEntityId({ memberId, videoId: videoId.toString() }) },
+  })
+
+  // load video reaction count by reaction type (LIKE/UNLIKE)
+  const reactionsCountByReactionType = await getOrCreateVideoReactionsCountByReactionId(store, video, reactionResult)
+
+  if (previousReactionByMember) {
```

Comment:
Let's avoid [Hadouken code](https://external-preview.redd.it/6TTm4tS5pMCiq2KlgviubpGKt1QzvetOi2Z6Z3lB1Nc.jpg?auto=webp&s=c1df961d84c8d76048a93c1d261e425d3737d779) (excesive indent) here. Create helper functions and flatten the structure. Also, some parts like `reactionsCountByReactionType.count` incrementing can be generalized.

Instead of
```
if (previousReactionByMember) {
  if (reactionResult === previousReactionByMember.reaction) {
    ...
  } else {
    ...
  }
} else {
  ...
}
```
let's have something like
```
myHelperFunction(video, ...)
...

function myHelperFunction(video, ...) {
  if (previousReactionByMember) {
    myHelperFunction2(...)
    return
  }
  
  // !previousReactionByMember
  ...
}

```